app/models/payments/OpenPaymentsModel.scala

```diff
@@ -18,10 +18,148 @@ package models.payments
 
 import java.time.LocalDate
 
-case class OpenPaymentsModel(paymentType: String,
-                             amount: BigDecimal,
-                             due: LocalDate,
-                             start: LocalDate,
-                             end: LocalDate,
-                             periodKey: String,
-                             overdue: Boolean = false)
+import play.api.i18n.Messages
+import play.api.libs.json.{Json, Writes}
+import views.templates.formatters.dates.DisplayDateRangeHelper.displayDateRange
+
+
+sealed trait OpenPaymentsModel {
+  val chargeType: ChargeType
+  val amount: BigDecimal
+  val due: LocalDate
+  val overdue: Boolean
+  val periodKey: String
+
+  def whatYouOweDescription(implicit messages: Messages): String =  chargeType match {
+    case OADebitCharge => messages("openPayments.officersAssessment")
+    case OADefaultInterestCharge => messages("openPayments.oaDefaultInterest")
+    case OAFurtherInterestCharge => messages("openPayments.oaFurtherInterest")
+    case OACharge => messages("openPayments.vatOfficersAssessment")
+    case BnpRegPost2010Charge => messages("openPayments.vatBnpRegPost2010")
+    case FtnMatPre2010Charge => messages("openPayments.vatFtnMatPre2010")
+    case FtnMatPost2010Charge => messages("openPayments.vatFtnMatPost2010")
+    case MiscPenaltyCharge => messages("openPayments.vatMiscPenalty")
+    case FtnEachPartnerCharge => messages("openPayments.vatFtnEachpartner")
+    case MpPre2009Charge => messages("openPayments.vatMpPre2009")
+    case MpRepeatedPre2009Charge => messages("openPayments.vatMpRepeatedPre2009")
+    case CivilEvasionPenaltyCharge => messages("openPayments.vatCivilEvasionPenalty")
```

Comment:
Please add a default case here (i.e. `_` and throw an `IllegalArgumentException`)